### DISCUSSION TITLE
Nested InnerJoin Pocketbase as a framework

### DISCUSSION THREAD

**Discussion by Clovis-krz** [pocketbase/pocketbase#2942] [Q&A] [answered]
Hi, i am trying to access a nested relation in a search query using InnerJoin. 
In my example, i want to return a table1 row if table1.availability.monday is true knowing that monday is called a relation in availability pointing to the table day_availability and morning is a boolean field in day_availability. These are relations pointing to the other table as a single relation. Here is the code that is not working
```
queryErr := app.Dao().RecordQuery(collection).
		InnerJoin("availability as availability", dbx.HashExp{}).
		InnerJoin("day_availability as monday", dbx.HashExp{"availability.monday.morning": true}).
		All(&records)
```
I hope my explanation is clear enough.

**Comment by ganigeorgiev** [comment]
I don't understand the above example, nor the described db structure. How `day_availability` is related to `availability`? Could you provide a simple ERD or an example dataset?

`InnerJoin`, `LeftJoin`, etc. methods operates in the same way as their SQL equivalents, aka. you need to specify an `ON` expression based on which to perform the join.

**Comment by Clovis-krz** [comment]
Sorry for the confusion,
Here are the collections : 
- table1
- availability
- day_availability

We can access availability from the field availability in table1 (single relation)
We can access day_availability from the field monday in availability (single relation)
morning is a bool field in day_availability

So that we can access morning the bool value using table1.availability.monday.morning (if we were using dart sdk for example)

How would i use ON expression ? As a dbx.NewExp() ? I am confused since i never used SQL with relations before ( I am more familiar with ORM's)

**Reply by ganigeorgiev** [reply, marked as answer]
Yes, `dbx.NewExp` is one way to define a raw `ON` clause. Here is an example:

```go
app.Dao().RecordQuery(collection).
        InnerJoin("availability a", dbx.NewExp(fmt.Sprintf("[[a.id]] = [[%s.availability]]", collection.Name))).
        InnerJoin("day_availability da", dbx.NewExp("[[da.id]] = [[a.monday]]")).
        AndWhere(dbx.HashExp{"da.morning": true}).
        All(&records)
```
_(the brackets are just for quotes to ensure that the columns don't collide with some reserved keyword)_

**Reply by Clovis-krz** [reply]
Its works ! Thanks so much !
